src/start_server.ts

```diff
@@ -181,6 +161,179 @@ function startWithPort(userOptions: ServerOptions) {
       });
     }
   }
+}
+
+/**
+ * Determines whether a protocol requires HTTPS
+ * @param {string} protocol Protocol to evalulate.
+ * @returns {boolean}
+ */
+function isHttps(protocol: string) {
+  return ['https/1.1', 'https', 'http2', 'h2'].indexOf(protocol) > -1;
+}
+
+/**
+ * Gets the URLs for the main and component pages
+ * @param {ServerOptions} options
+ * @returns {{serverUrl: {protocol: (string|string), hostname: string, port: string}, componentUrl: url.Url}}
+ */
+function getServerUrls(options: ServerOptions) {
+  let serverUrl = {
+    protocol: isHttps(options.protocol) ? 'https' : 'http',
+    hostname: options.hostname,
+    port: `${options.port}`,
+  };
+  let componentUrl: url.Url = Object.assign({}, serverUrl);
+  componentUrl.pathname = `components/${options.packageName}/`;
+  return {serverUrl, componentUrl};
+}
+
+/**
+ * Handles server-ready tasks (prints URLs and opens browser)
+ * @param {ServerOptions} options
+ */
+function handleServerReady(options: ServerOptions) {
+  let {serverUrl, componentUrl} = getServerUrls(options);
```

Comment:
Mainly because node 4 supports nearly all the ES6 we want to use, except destructuring, and TypeScript only supports async/await when targeting ES6. Unfortunately we can't tell TypeScript to compile out just rest/spread, and we don't want to complicate things with a Closure/Babel pass after TypeScript.